internal/testutil/mig/test_case.go

```diff
@@ -0,0 +1,71 @@
+package mig
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/stretchr/testify/require"
+)
+
+func CreateAndRunTest(t *testing.T, test *resource.TestCase) {
+	t.Helper()
+	resource.ParallelTest(t, CreateTest(t, test))
+}
+
+func CreateTestAndRunUseExternalProvider(t *testing.T, test *resource.TestCase, externalProviders, additionalProviders map[string]resource.ExternalProvider) {
+	t.Helper()
+	resource.ParallelTest(t, CreateTestUseExternalProvider(t, test, externalProviders, additionalProviders))
+}
+
+// CreateTest returns a new TestCase that reuses step 1 and adds a TestStepCheckEmptyPlan.
+// Requires: `MONGODB_ATLAS_LAST_VERSION` to be present.
+func CreateTest(t *testing.T, test *resource.TestCase) resource.TestCase {
+	t.Helper()
+	validateReusableCase(t, test)
+	firstStep := test.Steps[0]
+	steps := []resource.TestStep{
+		useExternalProvider(&firstStep, ExternalProviders()),
+		TestStepCheckEmptyPlan(firstStep.Config),
+	}
+	newTest := reuseCase(test, steps)
+	return newTest
+}
+
+// CreateTestUseExternalProvider returns a new TestCase that reuses step 1 and adds a TestStepCheckEmptyPlan with the additionalProviders.
+// Requires: `MONGODB_ATLAS_LAST_VERSION` to be present.
+// externalProviders: e.g., ExternalProvidersWithAWS() or ExternalProviders("specific_sem_ver").
+// additionalProviders: e.g., acc.ExternalProvidersOnlyAWS(), can also be nil.
+func CreateTestUseExternalProvider(t *testing.T, test *resource.TestCase, externalProviders, additionalProviders map[string]resource.ExternalProvider) resource.TestCase {
```

Comment:
i see you check it's not presente here?  "require.NotContains..."

Reply:
The `reuseCase`function  creates a new `resources.TestCase`. This new TestCase will never have the `ProtoV6ProviderFactories` set.

Reply:
> i see you check it's not presente here? "require.NotContains(t, additionalProviders,..."

This is to avoid a case where the caller of CreateTestUseExternalProvider would use `ExternalProvidersWithAWS()` as I'm using TestStepCheckEmptyPlan which will always use `ProtoV6ProviderFactories` for the `mongodbatlas` provider